### PR TITLE
rinad: ipcm: implement plugin_load_kernel()

### PR DESCRIPTION
**Pull request by vmaffione**
This patch allows ipcm to modprobe/rmmod plugin modules.

Maintainers: @msune

### CODE REVIEW

File: rinad/src/ipcm/ipcm.cc [hunk 1]
```diff
@@ -1151,6 +1154,69 @@ IPCManager_::select_policy_set(Addon* callee, Promise* promise,
...
+		result = IPCM_FAILURE;
+
+	} else if (pid == 0) {
+		// child
+		if (load) {
+			execl("modprobe", "modprobe", plugin_name.c_str(), NULL);
+			execlp("modprobe", "modprobe", plugin_name.c_str(), NULL);
+
+		} else {
+			execl("rmmod", "rmmod", plugin_name.c_str(), NULL);
```
Comment (sandervrijders) [inline rinad/src/ipcm/ipcm.cc]: Shouldn't this be modprobe -r?

Comment (vmaffione) [inline rinad/src/ipcm/ipcm.cc]: Is it not the same? For me it's ok to switch "modprobe -r".

Comment (sandervrijders) [inline rinad/src/ipcm/ipcm.cc]: Well, modprobe handles module dependencies as well. modprobe -r: If the modules it depends on are also unused, modprobe will try to remove them too.

I always see it as:
insmod / rmmod
modprobe / modprobe -r

**Comment by vmaffione** [comment]
It's not an external API call, it's only internally used by the plugin_load() method. And it's intended to be used only in this way. plugin_load() methods hides the detail on whether the plugin is for kernel space code or userspace code.

**Comment by vmaffione** [comment]
There is indeed. Open up the IPCM console and

>> plugin-load 4 red-pcc
>> plugin-unload 4 red-pcc

Check kernel logs to see that things really happen.

**Comment by msune** [comment]
@vmaffione OK. I got confused by the Addon callee.

But you don't need the callee nor the promise, because the API is synchronous, so please remove them. 

We will have namespace problems with user-space kernel space naming of the policies. I suggest using something like:

`kernel:<policy_name>`
`user:<policy_name>`

We could even go beyond this and have a fully qualified name:

`kernel:<component>:<sub_component*>:..:<policy_name>`
`user:<component>:<sub_component*>:..:<policy_name>`

**Comment by vmaffione** [comment]
@msune: We can choose to have that "kernel:" "user:" approach for namespaces, but this requires discussion (not in this PR).

**Comment by vmaffione** [comment]
@msune : was that an ACK or not?

**Comment by msune** [comment]
No it wasn't.  `040dc7a` partially addresses this, but:

* Callee is not needed by plugin_load_kernel()
* The parameterl `bool load` is strange; wouldn't it make more sense to have a plugin_load_kernel() and a plugin_unload_kernel()?

**Comment by vmaffione** [comment]
Removed the callee argument as suggested, thanks.

Regarding the plugin_load_kernel() thing, it makes sense to have one in order to reuse code. It is the same technique already used in plugin_load() in IPCM and IPCP. So I would not change that in this PR.

**Comment by edugrasa** [comment]
ACK, verified working

**Comment by edugrasa** [comment]
@msune are you for an ACK or not convinced yet? :)

File: rinad/src/ipcm/ipcm.cc [hunk 2]
```diff
@@ -1151,6 +1154,55 @@ IPCManager_::select_policy_set(Addon* callee, Promise* promise,
...
+		ss << "Kernel plugin (un)loading: fork() failed";
+		FLUSH_LOG(ERR, ss);
+		result = IPCM_FAILURE;
+
+	} else if (pid == 0) {
+		// child
+		if (load) {
+			execl("modprobe", "modprobe", plugin_name.c_str(),
+			      NULL);
+			execlp("modprobe", "modprobe", plugin_name.c_str(),
```
Comment (msune) [inline rinad/src/ipcm/ipcm.cc]: Why 2 calls to exec? execlp should be enough I guess

(Commented also in the other PR, sorry for duplicating)

**Comment by msune** [comment]
Once this comment about exec(s) is addressed, for me it would be an ACK.

**Pull request merged**